layered-architectures-module-2/Carting.DAL/Repositories/MongoRepositoryBase.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Carting.DAL.Models;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Carting.DAL.Repositories;
+
+public class MongoRepositoryBase<T> : IMongoRepository<T> where T:EntityBase
+{
+    protected readonly IMongoDatabase _mongoDatabase;
+    protected readonly IMongoCollection<T> _itemCollection;
+    public MongoRepositoryBase(IMongoDatabase mongoDatabase)
+    {
+        _mongoDatabase = mongoDatabase;
+        _itemCollection = _mongoDatabase.GetCollection<T>(GetCollectionName());
+    }
+    
+    public virtual Task CreateAsync(T obj)
+    {
+        return _itemCollection.InsertOneAsync(obj);
+    }
+
+    public virtual Task UpdateAsync(T obj)
+    {
+        return _itemCollection.ReplaceOneAsync(entity => entity.Id == obj.Id, obj);
+    }
+    
+    public virtual Task GetByIdAsync(string id)
+    {
+        return _itemCollection.FindAsync(entity => entity.Id == id);
```

Comment:
JFYI,
here is a holy war topic, I'd like to share some thoughts with you for your awareness. You DON'T need to do anything here. just an additional topic.
https://www.youtube.com/watch?v=Q2zDatDVnO0
https://stackoverflow.com/questions/38017016/async-task-then-await-task-vs-task-then-return-task